stdlib/public/core/StringGuts.swift

```diff
@@ -244,859 +248,508 @@ extension _StringGuts {
+      Builtin.reinterpretCast(object))
+    self._objectBitPattern |= _smallBit | _twoByteCodeUnitBit
+    _sanityCheck(_isSmallCocoa)
+    _invariantCheck()
   }
+}
 
-  //
-  // Cocoa (non-tagged) Strings
-  //
-  ///*fileprivate*/ internal // TODO: private in Swift 4
+extension _StringGuts {
   @_inlineable
-  public // @testable
-  var _isNonTaggedCocoa: Bool {
-    @inline(__always)
-    get { return _isNonTaggedObjCPointer(_object) }
-  }
-
-  /*fileprivate*/ internal // TODO: private in Swift 4
-  var _cocoa: NonTaggedCocoaString? {
-    guard _isNonTaggedCocoa else { return nil }
-    return NonTaggedCocoaString(
-      _bridgeObject(toNonTaggedObjC: _object),
-      isSingleByte: self.isSingleByte,
-      start: UnsafeRawPointer(bitPattern: _otherBits))
-  }
-
-  /*fileprivate*/ internal // TODO: private in Swift 4
-  init(_ s: NonTaggedCocoaString) {
-    _sanityCheck(!_isObjCTaggedPointer(s.owner))
-    self.init(
-      _unflagged: _bridgeObject(fromNonTaggedObjC: s.owner),
-      isSingleByte: s.isSingleByte,
-      otherBits: UInt(bitPattern: s.start))
-  }
-
-  //
-  // Unsafe Strings
-  //
-
-  ///*fileprivate*/ internal // TODO: private in Swift 4
-  public // TODO(StringGuts): for testing only
-  var _isUnsafe: Bool {
-    return _isTagged && _objectBitPattern & _smallBit == 0
-  }
-
   @_versioned
-  var _unsafeString: UnsafeString? {
-    guard _isUnsafe else { return nil }
-    return _asUnsafe
+  internal
+  var _unmanagedASCIIView: _UnmanagedString<UInt8> {
+    _sanityCheck(isASCII)
+    if _isUnmanaged {
+      return _asUnmanaged()
+    } else if _isNative {
+      return _nativeStorage(of: UInt8.self).unmanagedView
+    } else {
+      _sanityCheck(_isContiguousCocoa)
+      return _asContiguousCocoa(of: UInt8.self)
+    }
   }
 
+  @_inlineable
   @_versioned
   internal
-  var _asUnsafe: UnsafeString {
-    _sanityCheck(_isUnsafe)
-    // Unflag it, untag it, and go
-    let pointer = UnsafeMutableRawPointer(
-      bitPattern: self._untaggedUnflaggedBitPattern
-    )._unsafelyUnwrappedUnchecked
-    let count = Int(bitPattern: self._otherBits)
-    _sanityCheck(count >= 0)
-    return UnsafeString(
-      baseAddress: pointer,
-      count: count,
-      isSingleByte: self.isSingleByte)
+  var _unmanagedUTF16View: _UnmanagedString<UTF16.CodeUnit> {
+    if _isUnmanaged {
+      return _asUnmanaged()
+    } else if _isNative {
+      return _nativeStorage(of: UTF16.CodeUnit.self).unmanagedView
+    } else if _isContiguousCocoa {
+      return _asContiguousCocoa(of: UTF16.CodeUnit.self)
+    } else {
+      _sanityCheckFailure("String isn't contiguous")
+    }
   }
 
   @_inlineable
   @_versioned
-  init(_ s: UnsafeString) {
-    // Tag it, flag it, and go
-    let object = _bridgeObject(taggingPayload: UInt(bitPattern: s.baseAddress))
-    self.init(
-      _unflagged: object,
-      isSingleByte: s.isSingleByte,
-      otherBits: UInt(s.count))
+  internal
+  var _unmanagedRawBufferView: UnsafeRawBufferPointer {
```

Comment:
Just curious, who would want to use this?

Reply:
It was useful as a temporary helper while I had both UnsafeString and _UnmanagedString<T>. It seems pointless to keep it now.